### PR TITLE
CSF3: Add auto-title support to on-demand V7/V6 refactor

### PR DESCRIPTION
**Pull request by shilman**
Issue: #15714 #16059 #16075

## What I did

Refactored to add auto-title to:
- [x] Story store V7 mode
- [x] Story store v6 compatibility mode

Also:
- [x] Removed spurious warnings
- [x] Deprecated `previewCsfV3` feature flag

## How to test

See attached `react-ts` example

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: lib/store/src/autoTitle.ts [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+interface StoriesSpecifier {
+  directory: string;
+  files?: string;
+  titlePrefix?: string;
 }
 
-const autoTitleV2 = (meta: Meta, fileName: string) => {
-  return meta.title;
-};
+interface NormalizedStoriesEntry {
```
Comment (tmeasday) [inline lib/store/src/autoTitle.ts:13]: Are we going to rename this type?

File: lib/client-api/src/StoryStoreFacade.ts [hunk 2]
```diff
@@ -129,7 +133,10 @@ export class StoryStoreFacade<TFramework extends AnyFramework> {
     this.clearFilenameExports(fileName);
 
     const { default: defaultExport, __namedExportsOrder, ...namedExports } = fileExports;
-    const { id: componentId, title } = defaultExport || {};
+    // eslint-disable-next-line prefer-const
+    let { id: componentId, title } = defaultExport || {};
+
+    title = title || autoTitle(fileName, STORIES);
```
Comment (tmeasday) [inline lib/client-api/src/StoryStoreFacade.ts:139]: Can you add a test for this?

**Review by tmeasday — COMMENTED** [review]
Generally LGTM, two things.

File: lib/client-api/src/StoryStoreFacade.ts (hunk 2 above)
Comment (shilman) [reply lib/client-api/src/StoryStoreFacade.ts:139]: done

File: lib/store/src/autoTitle.ts (hunk 1 above)
Comment (shilman) [reply lib/store/src/autoTitle.ts]: done

**Comment by shilman** [comment]
Self-merging @tmeasday

**Pull request merged**